src/test/java/api/loans/DueDateScheduledNoticesTests.java

```diff
@@ -621,4 +631,138 @@ public void scheduledOverdueNoticesShouldBeDeletedAfterOverdueFineIsCharged() {
+    scheduledNoticeProcessingClient.runLoanNoticesProcessing(dueDate.plusDays(1));
+
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(scheduledNoticesClient::getAll, empty());
+  }
+
+  @Test
+  public void scheduledOverdueNoticesShouldBeDeletedAfterDueDateChange() {
+    NoticePolicyBuilder noticePolicy = createSendNoticePolicy();
+    use(noticePolicy);
+    IndividualResource loan = createScheduledNotice(new DateTime(2020, 1, 1, 12, 0, 0, UTC));
+
+    var dueDate = getDateTimeProperty(loan.getJson(), "dueDate");
+    scheduledNoticeProcessingClient.runLoanNoticesProcessing(dueDate);
+
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(scheduledNoticesClient::getAll, hasSize(1));
+    assertThat(scheduledNoticesClient.getAll().get(0).getString("nextRunTime"),
+      is(dueDate.plusMinutes(5).toString()));
+
+    changeDueDateFixture.changeDueDate(new ChangeDueDateRequestBuilder()
+      .forLoan(loan.getId())
+      .withDueDate(dueDate.plusWeeks(2)));
+
+    scheduledNoticeProcessingClient.runLoanNoticesProcessing(dueDate.plusDays(1));
+
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(scheduledNoticesClient::getAll, hasSize(1));
+
+    assertThat(scheduledNoticesClient.getAll().get(0).getString("nextRunTime"),
+      is(dueDate.plusWeeks(2).plusMinutes(5).toString()));
+  }
+
+  @Test
+  public void scheduledOverdueNoticesShouldBeDeletedAfterRenew() {
+    NoticePolicyBuilder noticePolicy = createSendNoticePolicy();
+    use(noticePolicy);
+    IndividualResource user = usersFixture.steve();
+    ItemResource item = itemsFixture.basedUponNod();
+    IndividualResource loan = createScheduledNotice(item, user, now());
+
+    loansFixture.renewLoan(item, user);
+    var dueDate = getDateTimeProperty(loan.getJson(), "dueDate");
+    scheduledNoticeProcessingClient.runLoanNoticesProcessing(dueDate.plusDays(1));
+
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(scheduledNoticesClient::getAll, empty());
+  }
+
+  private NoticePolicyBuilder createSendNoticePolicy() {
+    UUID templateId = UUID.randomUUID();
+    JsonObject loanNotice = new NoticeConfigurationBuilder()
+      .withTemplateId(templateId)
+      .withDueDateEvent()
+      .withAfterTiming(TIMING_AFTER)
+      .recurring(TIMING_RECURRING)
+      .sendInRealTime(true)
+      .create();
+    templateFixture.createDummyNoticeTemplate(templateId);
+
+    NoticePolicyBuilder noticePolicy = new NoticePolicyBuilder()
+      .withName("loan policy")
+      .withLoanNotices(Collections.singletonList(loanNotice));
+    use(noticePolicy);
+    return noticePolicy;
+  }
+
+  private IndividualResource createScheduledNotice(DateTime loanDate) {
+    return createScheduledNotice(itemsFixture.basedUponNod(), usersFixture.steve(), loanDate);
+  }
+
+  private IndividualResource createScheduledNotice(ItemResource item, IndividualResource user,
```

Comment:
Think of a better name. This method itself does not create notices.

Reply:
renamed